### DISCUSSION TITLE
License is not applicable to obfuscated code

### DISCUSSION THREAD

**Discussion by adri326** [ThioJoe/YT-Spammer-Purge#436] [General]
The `prepare_modes.py` file contains [a bunch of obfuscated data](https://github.com/ThioJoe/YT-Spammer-Purge/blob/main/Scripts/prepare_modes.py#L313-L359).

However, the GPLv3 license define "source code" as ["the preferred form of the work for making modifications to it"](https://github.com/ThioJoe/YT-Spammer-Purge/blob/main/LICENSE#L114-L116).
This [rules out](https://opensource.stackexchange.com/questions/5759/code-obfuscation-and-gpl3) any form of obfuscated code if it isn't directly programmed in the obfuscated form.

This means (to me at least) that any copy of this script made by someone other than you and that doesn't remove/deobfuscate this section from `prepare_modes.py` would not be covered by the terms of the license.

Ethically speaking, why make something only partially freely available and then label it as "free and open source"? Trusting other's code is already a difficult task, so why make it even harder by obfuscating that code?

I don't know what to end this with, other than to say that this isn't legal advice and that your work is very valuable.

**Comment by ThioJoe** [comment]
The code and logic isn't obfuscated, only certain keyword strings / variable values, and not even for all the filters. And I use the term 'obfuscated' loosely. Basically only to the degree that it prevents some moron scammer from being able to take a quick glance at the code and know some of the hardcoded keywords. Anyone who knows anything about programming can probably figure out how to see what they are.

**Reply by fdabek1** [reply]
> The code and logic isn't obfuscated, only certain keyword strings / variable values, and not even for all the filters. And I use the term 'obfuscated' loosely. Basically only to the degree that it prevents some moron scammer from being able to take a quick glance at the code and know some of the hardcoded keywords. Anyone who knows anything about programming can probably figure out how to see what they are.

After looking at the code, it definitely is not obfuscated in the way that Linus made it out to be.  Anybody can easily reverse it, which probably for the benefit of contributions making it not obfuscated may be better.  Look at something like uBlock: all of the filter lists are public and not "obfuscated".  Spam problems is always a constant battle no matter what.

Looking into a network such as this [Roberta fine-tuned one](https://huggingface.co/mvonwyl/roberta-twitter-spam-classifier) would greatly make this better and more "obfuscated".

**Reply by ThioJoe** [reply]
Yea I'll consider it, there might need to be some longer term adjustments if (when) spammers start to compensate.
